### DISCUSSION TITLE
Send Text to Cargo

### DISCUSSION THREAD

**Discussion by ChuckStarchaser** [endless-sky/endless-sky#7529] [Ideas]
This is what I think a very tiny improvement to make.  When a mission asks you to recover something, I heard that there is currently no way to make the object appear in the cargo.  So, for example, this mission was to recover an alien probe that had landed on a human planet and had been bought by a pirate.  I recovered it, and before bringing it back I landed on a planet and went to see my cargo in Outfitters and Trading, but I saw nothing there, and thought the mission was lost.  Apparently, there is currently no way for mission scripting to add something to the cargo hold.
My idea for a solution, though I'm not familiar with your sources, is for cargo to be able to receive a string from mission scripting.  So, something like,

```
 void add_string_to_cargo( char const * not_too_long_pls );
 void delete_cargo_string();
```

That the scripting language can call.
The string would then show at the bottom left of the Trading screen, same place where it often shows how many outfits are there, rather in addition to anything that might be shown there.  Thus,

```
  add_string_to_cargo( "Alien probe" );
```

Should result in the bottom of the Trading screen reading, for example,

```
  22 tons of outfits.  Alien probe.                    free:        0
```

Motivation:
Otherwise the player might land at a planet, not see the probe that "should have been in the cargo" and panic that a bug in the code made it disappear, and think the mission is lost.

Motivation for NOT suggesting an actual object in cargo:
It is simpler to just add a string.  If there was a way to add an item to cargo, you need to worry about what to do if there is no cargo space, etcetera, etcetera,  ...

**Comment by SirTechSpec** [comment]
I think it makes more sense to have an actual object in cargo, so you *do* need to worry about cargo space, losing the ship that's carrying it, etc. etc. 

For now, it looks like most retrieval missions are just putting the cargo on your ship when you accept the mission. As a player, I just wrote this off as "reserving space" to guarantee you'd be able to complete the mission successfully. It's not ideal, but IMO preferable to what you're proposing, if we're not going to do the thing properly and implement retrieving cargo at waypoints.

**Reply by ChuckStarchaser** [reply]
I agree with your point about what makes more sense.  The problems it causes are numerous, however:  What if the retreival mission's attempt to reserve space in your cargo fails because it is already full?  How do you justify to the player denying the mission on the basis of failing to occupy space in the cargo preemptively?  What if you take the mission and then change ships for a faster one without sufficient cargo space?  What if you use vitalchip's trick to get around the cargo selling message when leaving with a smaller ship by leaving with two or three other ships, telling them to Hold ground, landing in another system, and remotely parking them?  Now you've managed to defeat the reserved space trick by shoving it onto another ship in your fleet.
  This is not the first game to face this problem.  Wing Commander's Privateer faced the exact same problem, so what they did was to simply ignore maximum tonnage of your ship, such that you could load it to the brim and then take additional cargo missions AND retrieve things in space, all of which could add to like a 20% overload.  It's not elegant, but then again, anything else is threatened by the fact that most players will always optimize profit by filling their cargoes to the brim.  Another solution, far more reasonable perhaps, would be to make cargo retrieval space separate from official cargo space, and specify how much retrieval space is there in each ship model.  You'd not be able to fill this space at the trading room;  only in space.
  Anyways, my suggested "text-to-cargo" solution was meant as a temporary measure while working on a better solution.  I should have said so;  I know it is not elegant at all.

EDIT:  Submitted feature request 7758 for separate retrieval space.
